### PR TITLE
Update fetch contract for aggregated limit

### PR DESCRIPTION
**Pull request by lrahmani**
- Add reverse aggregate allowance
- Make owner a regular admin

### CODE REVIEW

File: fetch/cosmwasm_contract/src/contract.rs [hunk 1]
```diff
@@ -685,10 +803,10 @@ fn _only_original_owner<S: Storage>(env: &Env, storage: &S) -> HandleResult {
         Err(StdError::unauthorized())
     }
 }
+*/
```
Comment (pbukva) [inline fetch/cosmwasm_contract/src/contract.rs:806]: you can ditch this commented-out code.

File: fetch/cosmwasm_contract/src/contract.rs [hunk 2]
```diff
@@ -659,7 +776,7 @@ fn verify_refund_swap_id<S: Storage>(id: u64, storage: &S) -> HandleResult {
  * ***************************************************/
 
 fn only_admin<S: Storage>(env: &Env, storage: &S) -> HandleResult {
-    _only_role(&AccessRole::Admin, env, storage).or(_only_original_owner(env, storage))
+    _only_role(&AccessRole::Admin, env, storage) //.or(_only_original_owner(env, storage))
```
Comment (pbukva) [inline fetch/cosmwasm_contract/src/contract.rs:779]: you can ditch commented-out code.

File: fetch/cosmwasm_contract/src/lib.rs [hunk 3]
```diff
@@ -2,6 +2,7 @@ pub mod contract;
 pub mod msg;
 pub mod state;
 pub mod access_control;
+pub mod error;
```
Comment (pbukva) [inline fetch/cosmwasm_contract/src/lib.rs:5]: Is this required?

**Review by pbukva — CHANGES_REQUESTED** [review]
Looks good. Requsting changes only so the you would have a quick look at my inline comments.

File: fetch/cosmwasm_contract/src/lib.rs (hunk 3 above)
Comment (lrahmani) [reply fetch/cosmwasm_contract/src/lib.rs:5]: not for this branch/pr

File: fetch/cosmwasm_contract/src/contract.rs (hunk 2 above)
Comment (lrahmani) [reply fetch/cosmwasm_contract/src/contract.rs:779]: sure, I kept it because I was not sure about this

**Review by pbukva — APPROVED** [review]
LGTM

**Review by pbukva — APPROVED** [review]
LGTM

**Pull request merged**